### PR TITLE
test: app components

### PR DESCRIPTION
**Pull request by martinharperlee**
## Summary

Added tests to cover some of the components with render/components/App

These components currently have no coverage, the aim was to increase this as much as possible, no changes to the main component files were required.

### CODE REVIEW

**Comment by martinharperlee** [comment]
This was my first PR for the Ark Eco System, please let me know if I have missed anything in the process for submitting this.

File: __tests__/unit/components/App/AppUpdater.spec.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+  it('should format percentage', () => {
+    expect(wrapper.vm.formattedPercentage).toBe('0.00%')
+    wrapper.vm.progressUpdate.percent = 5
+    expect(wrapper.vm.formattedPercentage).toBe('5.00%')
+    wrapper.vm.progressUpdate.percent = 5.56567688
+    expect(wrapper.vm.formattedPercentage).toBe('5.57%')
+  })
+
+  it('should return if Linux', () => {
```
Comment (brenopolanski) [inline __tests__/unit/components/App/AppUpdater.spec.js]: Change from:

`it('should return if Linux', () => {`

To:

`it('should return if it is linux', () => {`

File: __tests__/unit/components/App/AppUpdater.spec.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    })
+    createWrapper()
+    expect(wrapper.vm.isAppImage).toBe(false)
+
+    Object.defineProperty(process, 'env', {
+      value: originalEnv
+    })
+  })
+
+  it('should return title if not authorized downloading', () => {
```
Comment (brenopolanski) [inline __tests__/unit/components/App/AppUpdater.spec.js]: Change from:

`it('should return title if not authorized downloading', () => {`

To:

`it('should return the title if download is not authorized', () => {`

File: __tests__/unit/components/App/AppUpdater.spec.js [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    expect(wrapper.vm.title).toBe(undefined)
+  })
+
+  it('should return formatted release notes', () => {
+    /* eslint-disable no-useless-escape */
+    const newHTML = '<html><head></head><body><div><h3>Title</h3><span href=\"#\">Test Link</span></div></body></html>'
+    expect(wrapper.vm.releaseNotes).toBe(newHTML)
+  })
+
+  it('should return description image dependant on theme', () => {
```
Comment (brenopolanski) [inline __tests__/unit/components/App/AppUpdater.spec.js]: Change from:

`it('should return description image dependant on theme', () => {`

To:

`it('should return the image of the description depending on the theme', () => {`

File: __tests__/unit/components/App/AppFooter.spec.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { shallowMount } from '@vue/test-utils'
+import { AppFooter } from '@/components/App'
+
+let wrapper
+const translation = 'Mock Translaton'
```
[the remark this answers is not in the document]
Comment (martinharperlee) [reply __tests__/unit/components/App/AppFooter.spec.js]: Done. I'll pay better attention to my spelling next time.

File: __tests__/unit/components/App/AppSidemenu/AppSidemenu.spec.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+  $route: {
+    name: 'mock_route'
+  },
+  $router: {
+    push
+  },
+  $t: () => 'Mock Translaton',
+  $refs: {
+    settings: {
+      showSettings
```
[the remark this answers is not in the document]
Comment (martinharperlee) [reply __tests__/unit/components/App/AppSidemenu/AppSidemenu.spec.js]: Good spot

**Comment by martinharperlee** [comment]
All above comments addressed

**Comment by boldninja** [comment]
@martinharperlee Can you provide me with your ARK address so I can send you your bounty reward? You can also contact me on ARK's Slack - https://ark.io/slack (username: Rok [ARK Team]) and direct message me your address there.

**Pull request merged**